### PR TITLE
Refactor project structure. Fix Compile Remove / Compile Include

### PR DESCRIPTION
**Pull request by StefH**

### CODE REVIEW

**Comment by StefH** [comment]
@doboczyakos and @davidnemeti Can you take a look at this PR?

I still needed to fix some things.

**Comment by doboczyakos** [comment]
I think the build error is because of the portable-net40+win8+wpa81+wp8+sl5 and portable-net45+win8+wpa81+wp8 target frameworks. You should consider removing them. In my opinion no other modifications are needed.

**Comment by doboczyakos** [comment]
Changing back the LinqKit.Core namespace to LinqKit containing the AsExpandable extension will cause build error 'The call is ambiguous between the following methods or properties' if you have reference to both LinqKit.Core and LinqKit packages

**Comment by doboczyakos** [comment]
LinqKit.EntityFramework.Tests.Net452.DbAsyncTests.DbAsync_EnumerateShouldWorkAsync [FAIL]

This is because AsExpandable() was called from LinqKit.Core instead of LinqKit. That's why we moved it to a new namespace to avoid the 'ambiguous call' build error and stay compatible when someone calls AsExpandable on an EF or EF Core IQueryable.
In that rare case when someone used AsExpandable from LinqKit.Core this is a breaking change but it can be simply resolved by adding 'using LinqKit.Core;'

**Comment by StefH** [comment]
Build is build.

However these changes:
image


Are still needed, because you don;t need to include source files anymore because we now include a project (LinqKit.Core)

**Comment by doboczyakos** [comment]
Some files (ExpandableQuery, Extensions) are still needed that's why I've added #if NOEF to the beginning of appropriate files.

File: src/LinqKit.Microsoft.EntityFrameworkCore5/LinqKit.Microsoft.EntityFrameworkCore5.csproj [hunk 1]
```diff
@@ -22,26 +22,26 @@
...
     <PathMap>$(MSBuildProjectDirectory)=/</PathMap>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
   </PropertyGroup>
 
-  <ItemGroup>
+  <!--<ItemGroup>
     <Compile Remove="..\LinqKit.Core\*\AssemblyInfo.cs" />
     <Compile Include="..\LinqKit.Core\*.cs;..\LinqKit.Core\*\*.cs" Exclude="..\LinqKit.Core\*\AssemblyInfo.cs;bin\**;obj\**;**\*.xproj;packages\**" />
-  </ItemGroup>
+  </ItemGroup>-->
```
Comment (doboczyakos) [inline src/LinqKit.Microsoft.EntityFrameworkCore5/LinqKit.Microsoft.EntityFrameworkCore5.csproj:33]: At least Extensions.cs and ExpandableQuery.cs must be included

Comment (StefH) [reply src/LinqKit.Microsoft.EntityFrameworkCore5/LinqKit.Microsoft.EntityFrameworkCore5.csproj]: fixed

**Comment by StefH** [comment]
@doboczyakos 

I think all is correctly fixed, can you check ?

**Comment by doboczyakos** [comment]
> @doboczyakos
> 
> I think all is correctly fixed, can you check ?

Seems great. Thanks.

**Pull request merged**